internal/engine/configs_controller_test.go

```diff
@@ -0,0 +1,100 @@
+package engine
+
+import (
+	"context"
+	"os"
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/windmilleng/tilt/internal/model"
+	"github.com/windmilleng/tilt/internal/store"
+	"github.com/windmilleng/tilt/internal/testutils"
+	"github.com/windmilleng/tilt/internal/testutils/output"
+	"github.com/windmilleng/tilt/internal/testutils/tempdir"
+	"github.com/windmilleng/tilt/internal/tiltfile"
+)
+
+func TestConfigsController(t *testing.T) {
+	f := newCCFixture(t)
+	defer f.TearDown()
+
+	state := f.st.LockMutableStateForTesting()
+	m := model.Manifest{Name: "foo"}
+	mt := store.NewManifestTarget(m)
+	state.UpsertManifestTarget(mt)
+	state.PendingConfigFileChanges["Tiltfile"] = true
+	state.TiltfilePath = f.JoinPath("Tiltfile")
+	f.st.UnlockMutableState()
+
+	a := f.run()
+
+	expected := ConfigsReloadedAction{
+		Manifests:  []model.Manifest{{Name: "bar"}},
+		StartTime:  f.fc.Times[0],
+		FinishTime: f.fc.Times[1],
+	}
+
+	assert.Equal(t, expected, a)
+}
+
+func (f *ccFixture) run() ConfigsReloadedAction {
+	// configs_controller uses state.RelativeTiltfilePath, which is relative to wd
+	origDir, err := os.Getwd()
+	if err != nil {
+		f.T().Fatalf("error getting wd: %v", err)
+	}
+	err = os.Chdir(f.Path())
+	if err != nil {
+		f.T().Fatalf("error changing dir: %v", err)
+	}
+	defer func() {
```

Comment:
should this be part of the TearDown method?

alternatively, maybe all this code should just be an option on TempDirFixture

Reply:
> should this be part of the TearDown method?
I like it where it is, so that the state mutation and unmutation are in the same place. Also, so that the scope of the changed wd is minimized, and no other part of the test has to worry about what this function has done to global state.

I'd kind of prefer to fix our wd dependency rather than encoding it into TempDirFixture. I was tempted to do it for this PR, but I'm already 3 yaks deep.